### PR TITLE
Update summary description

### PR DESCRIPTION
**Pull request by kalletlak**
# Fixes
1. Update Summary description
2. Updated virtual study box default description
3. CSS updates in custom case selection popup

### CODE REVIEW

File: src/pages/studyView/StudyViewPage.tsx [hunk 1]
```diff
@@ -321,14 +317,109 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
                     </div>
                 )
             }
             return null;
         }
     }
 }
+
+
+interface IStudySummaryProps {
```
[the remark this answers is not in the document]
Comment (kalletlak) [reply src/pages/studyView/StudyViewPage.tsx]: we could just leave it for now as it's used only here

File: src/pages/studyView/StudyViewPage.tsx [hunk 2]
```diff
@@ -105,14 +109,10 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
             () => props.routing.location.query,
             query => {
                 if ('studyId' in query) {
-                    this.store.studyIds = (query.studyId as string).split(",");
+                    this.store.queriedIdsStr = (query.studyId as string);
```
Comment (alisman) [inline src/pages/studyView/StudyViewPage.tsx]: always check to see if the id is different before setting it.  mobx will fire even when you set a property with the same value and that will results in unnecessary re-render

File: src/pages/studyView/StudyViewPage.tsx [hunk 3]
```diff
@@ -105,14 +109,10 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
                 }
                 if ('id' in query) {
-                    this.store.studyIds = (query.id as string).split(",");
-                }
-                if (this.store.studyIds) {
-                    this.store.sampleAttrIds = ('sampleAttrIds' in query ? (query.sampleAttrIds as string).split(",") : []);
-                    this.store.patientAttrIds = ('patientAttrIds' in query ? (query.patientAttrIds as string).split(",") : []);
+                    this.store.queriedIdsStr = (query.id as string);
```
Comment (alisman) [inline src/pages/studyView/StudyViewPage.tsx]: what's difference between id and query?  (anyways, ditto comment above)

File: src/pages/studyView/StudyViewPage.tsx [hunk 4]
```diff
@@ -271,25 +271,18 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
-                            </If>
-                            {/*TDOD: currently show as Multiple Studies but should be shandles properly, i.e as in production*/}
-                            <If condition={this.store.queriedStudies!.length > 1}>
-                                <h3>Multiple Studies</h3>
-                            </If>
-                        </div>
-                    </div>
+                    <StudySummary
+                        studies={this.store.displayedStudies.result}
+                        originStudies={this.store.originStudies}
```
Comment (alisman) [inline src/pages/studyView/StudyViewPage.tsx]: is originStudies a promise?

File: src/pages/studyView/StudyViewPage.tsx [hunk 5]
```diff
@@ -362,14 +355,106 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
             )
         } else {
             //TODO: update with loading
-            if(this.store.filteredVirtualStudies.isComplete && !_.isEmpty(this.store.unknownQueriedIds)) {
+            if (this.store.filteredVirtualStudies.isComplete &&
```
Comment (alisman) [inline src/pages/studyView/StudyViewPage.tsx]: don't worry about loading behavior here.  i did some work on this in spa branch.

File: src/pages/studyView/StudyViewPage.tsx [hunk 6]
```diff
@@ -321,14 +317,109 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
+            if (lines.length > 1) {
+                //slice fist line as its already shown
+                return [<span style={{ whiteSpace: 'pre' }} dangerouslySetInnerHTML={{ __html: lines.slice(1).join('\n') }} />]
+            }
+        } else {
+            return _.map(this.props.studies, study => {
+                return (
+                    <span>
+                        <a
+                            href={buildCBioPortalUrl({ pathname: 'newstudy', query: { id: study.studyId } })}
```
[the remark this answers is not in the document]
Comment (alisman) [reply src/pages/studyView/StudyViewPage.tsx]: Yes. And SPA will change how we link between pages, so don't worry to much about this until after merge

File: src/pages/studyView/StudyViewPageStore.ts [hunk 7]
```diff
@@ -695,21 +694,34 @@ export class StudyViewPageStore {
...
+            return _.values(studies);
+        },
+        default: []
+    });
 
-            return Object.keys(studyIds);
+    // includes all physical studies from queried virtual studies
+    readonly queriedPhysicalStudyIds = remoteData({
+        await: () => [this.queriedPhysicalStudies],
+        invoke: async () => {
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: when we a mobxpromise is just a "facade" for another mobxpromise, i.e. there's no actualy async operation inside invoke, we should take asycn tag of method and return Promise.resolve(data).  A little clearer.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 8]
```diff
@@ -766,22 +778,73 @@ export class StudyViewPageStore {
...
+            if (this.filteredPhysicalStudies.result.length === 0 && this.filteredVirtualStudies.result.length === 1) {
+
+                const virtualStudy = this.filteredVirtualStudies.result[0]
+                return [{
+                    name: virtualStudy.data.name,
+                    description: virtualStudy.data.description,
+                    studyId: virtualStudy.id,
+                } as CancerStudy]
+            }
+            return this.queriedPhysicalStudies.result;
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: i would rather see a full if/else statement.   it's very easy to miss the return statement above.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 9]
```diff
@@ -766,22 +778,73 @@ export class StudyViewPageStore {
...
+        },
+        default: []
+    });
 
-        return [...this.filteredPhysicalStudies.result, ...filteredVirtualStudies];
-    }
+    //get originStudies when there is only one study in query and that too virtual study
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: we need a better description of what originStudies is in terms of business logic.  the name doesn't really tell me.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 10]
```diff
@@ -766,22 +778,73 @@ export class StudyViewPageStore {
...
+    readonly originStudies = remoteData({
+        await: () => [this.filteredPhysicalStudies, this.filteredVirtualStudies],
+        invoke: async () => {
+            if (this.filteredPhysicalStudies.result.length === 0 && this.filteredVirtualStudies.result.length === 1) {
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: this stuff is important business logic and we should have it tested.  so externalize this

File: src/pages/studyView/StudyViewPage.tsx [hunk 11]
```diff
@@ -362,14 +355,106 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
+                    </span>
+                )
+            })
+        }
+        return [];
+    }
+
+    render() {
+        return (
+            <div className={classnames("topBanner", styles.summary)}>
```
[the remark this answers is not in the document]
Comment (kalletlak) [reply src/pages/studyView/StudyViewPage.tsx]: its not. its a global style

File: src/pages/studyView/StudyViewPage.tsx (hunk 4 above)
Comment (kalletlak) [reply src/pages/studyView/StudyViewPage.tsx]: yes

File: src/pages/studyView/StudyViewPage.tsx [hunk 12]
```diff
@@ -362,14 +355,106 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
...
+            this.props.studies.length > 1 ||
+            this.props.studies[0].description.split(/\n/g).length > 1;
+    }
+
+    @computed get descriptionRemainingLines() {
+        if (this.props.studies.length === 1) {
+            const lines = this.props.studies[0].description.split(/\n/g);
+            if (lines.length > 1) {
+                //slice fist line as its already shown
+                return [<span style={{ whiteSpace: 'pre' }} dangerouslySetInnerHTML={{ __html: lines.slice(1).join('\n') }} />]
```
[the remark this answers is not in the document]
Comment (zhx828) [reply src/pages/studyView/StudyViewPage.tsx]: Yes for this section. The backend uses this way to hyperlink some information in the description.

File: src/pages/studyView/StudyViewPage.tsx (hunk 5 above)
Comment (kalletlak) [reply src/pages/studyView/StudyViewPage.tsx]: should I revert back these changes?

File: src/pages/studyView/StudyViewPage.tsx (hunk 2 above)
Comment (kalletlak) [reply src/pages/studyView/StudyViewPage.tsx]: will update it once https://github.com/cBioPortal/cbioportal-frontend/pull/1458 is merged

File: src/pages/studyView/StudyViewPage.tsx (hunk 5 above)
Comment (alisman) [reply src/pages/studyView/StudyViewPage.tsx]: no.  i can deal with it on merge

File: src/pages/studyView/StudyViewPage.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (alisman) [reply src/pages/studyView/StudyViewPage.tsx]: @jjgao just want to point out the script injection potential here.    it's your call whether this is something we ought to worry ourselves with.

**Pull request merged**